src/core/hooks/useRemoteList.ts

```diff
@@ -18,37 +18,72 @@ export default function useRemoteList<
     cacheKey?: string;
     isNecessary?: () => boolean;
     loader: () => Promise<DataType[]>;
+    staleWhileRevalidate?: boolean;
   }
 ): DataType[] {
   const dispatch = useAppDispatch();
   const loadIsNecessary = hooks.isNecessary?.() ?? shouldLoad(remoteList);
 
   const promiseKey = hooks.cacheKey || hooks.loader.toString();
-  const { cache } = usePromiseCache(promiseKey);
-
-  if (!remoteList || loadIsNecessary) {
-    const promise = Promise.resolve()
-      .then(() => {
-        dispatch(hooks.actionOnLoad());
-      })
-      .then(() => hooks.loader())
-      .then((val) => {
-        dispatch(hooks.actionOnSuccess(val));
-        return val;
-      })
-      .catch((err: unknown) => {
-        if (hooks.actionOnError) {
-          dispatch(hooks.actionOnError(err));
-          return null;
-        } else {
-          throw err;
-        }
-      });
-
-    cache(promise);
-
-    if (!remoteList?.items.length) {
-      throw promise;
+  const { cache, getExistingPromise } = usePromiseCache(promiseKey);
+  const staleWhileRevalidate = hooks.staleWhileRevalidate ?? true;
+
+  if (!remoteList) {
```

Comment:
This is causing a bug in the canvasser interface. The mistake is not here, but rather in the store. By default, a new list created with the `createList()` utility contains an empty array of items. An empty array is truthy, so with this new logic, the empty array is returned.

This causes the households2 UI to display as if there are no households, while the load is happening. If the load is slow, this could prompt users to create households anew, creating duplicates. IMO, this makes this bug a blocker for this PR, which is a bit unfair since it's not the fault of the PR really, but more of an architectural problem of not differentiating lists that have loaded as empty from yet-to-be-loaded lists.

One way of solving this would be to look at `loaded`, i.e. if `remoteList` is truthy but `remoteList.loaded` is `null`, we still load.

Reply:
This is what I think should work.

```suggestion
  if (!remoteList || !remoteList.loaded) {
```